test/Microsoft.TemplateEngine.Mocks/MockMountPointManager.cs

```diff
@@ -0,0 +1,56 @@
+using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Abstractions.Mount;
+using Microsoft.TemplateEngine.Mocks;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.TemplateEngine.Mocks
+{
+    /// <summary>
+    /// Mock for IMountPointManager interface to be used for unit testing
+    /// Can be set up to return failure on demanding certain mount points. The unavailable mountpoints to be defined in <see cref="UnavailableMountPoints"/>. 
+    /// </summary>
+    public class MockMountPointManager : IMountPointManager
+    {
+        public MockMountPointManager (IEngineEnvironmentSettings environmentSettings)
+        {
+            EnvironmentSettings = environmentSettings;
+        }
+        public IEngineEnvironmentSettings EnvironmentSettings
+        {
+            private set; get;
+        }
+
+        /// <summary>
+        /// Contains the list of mount points that are considered to be unavailable when demanding
+        /// </summary>
+        public List<MountPointInfo> UnavailableMountPoints { get; private set; } = new List<MountPointInfo>();
+
+        public void ReleaseMountPoint(IMountPoint mountPoint)
+        {
+            // do nothing
+        }
+
+        public bool TryDemandMountPoint(MountPointInfo info, out IMountPoint mountPoint)
+        {
+            mountPoint = new MockMountPoint(EnvironmentSettings);
```

Comment:
nit: Move to the end just before `return true` so the method doesn't return `false` and *and* a valid mount point, something the real implementation never does.